### PR TITLE
fix: update list endpoint and test cases

### PR DESCRIPTION
**Pull request by makram93**
List endpoint currently returns the flat docs which is wrong when we have chunks of documents. This PR solves it by removing the duplicates and only returning single docs for all chunks of the same parent.

### CODE REVIEW

File: tests/now_executors/NOWAnnLiteIndexer/test_executor.py [hunk 1]
```diff
@@ -79,28 +80,38 @@ def test_list(tmpdir, offset, limit):
...
+                'chunks_size': 2,
+            }
+            if has_chunk
+            else {
+                'offset': offset,
+                'limit': limit,
             }
-            if offset is not None
-            else {}
         )
```
Comment (joschkabraun) [inline tests/now_executors/NOWAnnLiteIndexer/test_executor.py:95]: I think we should change test to only add `offset` and `limit` in the case they are actually given / not `None`

File: tests/now_executors/NOWAnnLiteIndexer/test_executor.py [hunk 2]
```diff
@@ -79,28 +80,38 @@ def test_list(tmpdir, offset, limit):
...
         if l > 0:
-            assert list_res[0].id == str(offset) if offset is not None else '0'
-            assert list_res[0].uri == 'my-parent-uri'
-            assert len(list_res[0].chunks) == 0
-            assert list_res[0].embedding is None
-            assert list_res[0].text == ''
-            assert list_res[0].tags == {'parent_tag': 'value'}
+            if has_chunk:
+                assert len(list_res[0].chunks) == 0
+                assert len(set([d.id for d in list_res])) == l
```
Comment (joschkabraun) [inline tests/now_executors/NOWAnnLiteIndexer/test_executor.py:107]: I think `has_chunk` should also test for `tags`. So, we need to change `gen_docs` function

**Review by joschkabraun — COMMENTED** [review]
Nice fix!
I think we should change the behavior of the parameter as I outlined in the comments. Let me know what you think

File: now_executors/NOWAnnLiteIndexer/executor.py [hunk 3]
```diff
@@ -220,10 +222,34 @@ def list(self, parameters: dict = {}, **kwargs):
         - offset (int): number of documents to skip
         - limit (int): number of retrieved documents
         """
-
-        limit = int(parameters.get('limit', maxsize))
-        offset = int(parameters.get('offset', 0))
-        return self.da[offset : offset + limit]
+        limit = (
+            int(parameters.get('limit'))
+            if parameters.get('limit') is not None
```
[the remark this answers is not in the document]
Comment (makram93) [reply now_executors/NOWAnnLiteIndexer/executor.py:227]: I agree that `None` may not be the case but somehow the params passed is received as `float` inside the `/list` endpoint hence the type casting. Will update the test cases accordingly

File: tests/now_executors/NOWAnnLiteIndexer/test_executor.py (hunk 2 above)
Comment (makram93) [reply tests/now_executors/NOWAnnLiteIndexer/test_executor.py:107]: Thanks, have added the additional test check

**Review by joschkabraun — APPROVED** [review]
LGTM!

**Pull request merged**